http3/frames.go

```diff
@@ -184,3 +186,35 @@ func (f *settingsFrame) Append(b []byte) []byte {
 	}
 	return b
 }
+
+type goawayFrame struct {
+	ID quic.StreamID
+}
+
+func parseGoawayFrame(r io.Reader, l uint64) (*goawayFrame, error) {
+	if l > 8*(1<<10) {
+		return nil, fmt.Errorf("unexpected size for GOAWAY frame: %d", l)
+	}
+	buf := make([]byte, l)
+	if _, err := io.ReadFull(r, buf); err != nil {
+		if err == io.ErrUnexpectedEOF {
+			return nil, io.EOF
+		}
+		return nil, err
+	}
+	frame := &goawayFrame{}
+	b := bytes.NewReader(buf)
+	id, err := quicvarint.Read(b)
+	if err != nil {
+		return nil, err
+	}
+	frame.ID = quic.StreamID(id)
+	return frame, nil
```

Comment:
Need to handle the case where there’s unconsumed data in the reader.

Reply:
I thought about that, but too lazy at that moment 😄 .